moviebot/nlu/annotation/joint_bert/dataset.py

```diff
@@ -153,20 +155,26 @@ def _tokenize_and_label(
             index = text.find(slot_text)
             for word in text[start_idx:index].split():
                 labels.append(JointBERTSlot.to_index("OUT"))
-                labels.extend([_IGNORE_INDEX] * self._num_word_tokens(word) - 1)
+                labels.extend(
+                    [_IGNORE_INDEX] * self._num_inside_word_tokens(word)
+                )
 
             for i, word in enumerate(slot_text.split()):
-                labels.append(
-                    JointBERTSlot.to_index(
-                        ("B_" if i == 0 else "I_") + slot_label.upper()
-                    )
+                slot = (
+                    ("B" if i == 0 else "I")
+                    + ("_INQUIRE_" if intent == "INQUIRE" else "_PREFERENCE_")
+                    + slot_label.upper()
+                )
+
+                labels.append(JointBERTSlot.to_index(slot))
+                labels.extend(
+                    [_IGNORE_INDEX] * self._num_inside_word_tokens(word)
                 )
-                labels.extend([_IGNORE_INDEX] * self._num_word_tokens(word) - 1)
             start_idx = index + len(slot_text)
 
         for word in text[start_idx:].split():
             labels.append(JointBERTSlot.to_index("OUT"))
-            labels.extend([_IGNORE_INDEX] * self._num_word_tokens(word) - 1)
+            labels.extend([_IGNORE_INDEX] * self._num_inside_word_tokens(word))
```

Comment:
Just a clarification question, here you only keep the id of the first token for each word right?

Reply:
Yes. I have extended the docstrings with an example.